onnxruntime/test/contrib_ops/quantize_ops_test.cc

```diff
@@ -256,6 +281,60 @@ TEST(QuantizeLinearContribOpTest, QuantizeLinear_per_tensor_float_int8) {
   test.Run(OpTester::ExpectResult::kExpectSuccess, "", {kTensorrtExecutionProvider});
 }
 
+// Test uint16 com.microsoft.QuantizeLinear (per tensor)
+TEST(QuantizeLinearContribOpTest, QuantizeLinear_per_tensor_float_uint16) {
+  OpTester test("QuantizeLinear", 1, onnxruntime::kMSDomain);
+  std::vector<int64_t> dims{12};
+  test.AddInput<float>("x", dims, {
+                                      0.f, -128.f, 3.f, -3.f,  // rounding half to even
+                                      2.9f, -2.9f,             // round < .5
+                                      3.1f, -3.1f,             // round > .5
+                                      65536.f, -65534.f,       // critical point
+                                      70000.f, -70000.f        // saturate case
```

Comment:
Thanks! Credit to @yufenglee for creating the original 8bit Q/DQ tests with thought out cases. I copied and modified his original tests so that we're properly testing similar numerical edge-cases for 16bit Q/DQ.